click/_bashcomplete.py

```diff
@@ -97,6 +96,22 @@ def is_incomplete_argument(current_params, cmd_param):
         return True
     return False
 
+def get_user_autocompletions(ctx, args, incomplete, cmd_param):
+    """
+    :param ctx: context associated with the parsed command
+    :param args: full list of args
+    :param incomplete: the incomplete text to autocomplete
+    :param cmd_param: command definition
+    :return: all the possible user-specified completions for the param
+    """
+    choices = []
```

Comment:
Ah, hmm. I'm trying to avoid instantiation of a list here. More comments will follow.